utils/zerovec/src/ule/option.rs

```diff
@@ -0,0 +1,193 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use super::*;
+use core::mem::{self, MaybeUninit};
+
+/// This type is the [`ULE`] type for `Option<T>`.
+///
+/// # Example
+///
+/// ```rust
+/// use zerovec::ZeroVec;
+///
+/// let z = ZeroVec::alloc_from_slice(&[Some('a'), Some('á'), Some('ø'), Some('ł')]);
+///
+/// assert_eq!(z.get(2), Some(Some(('ø'))));
+/// ```
+// Invariants:
+// The MaybeUninit is zeroed when None (bool = false),
+// and is valid when Some (bool = true)
+#[repr(packed)]
+pub struct OptionULE<T>(bool, MaybeUninit<T>);
```

Comment:
Nit (optional): Consider changing the type parameter to `U` instead of `T`; I think we're not always consistent about this, but I like `T` to be the normal type and `U` to be the ULE type